### PR TITLE
roachprod: add `--gce-use-spot`

### PR DESCRIPTION
**Pull request by srosenberg**
Previously, `--gce-preemptible` was available.
This change adds an option to create a GCE spot
instance, whose lifetime can extend 24h; otherwise, it's essentially equivalent to a GCE preemptible.
VM metadata and billing estimator are updated to
handle both preemptible and spot instances.

Epic: none

Release note: None

### CODE REVIEW

File: pkg/roachprod/vm/gce/gcloud.go [hunk 1]
```diff
@@ -809,7 +824,10 @@ func (o *ProviderOpts) ConfigureCreateFlags(flags *pflag.FlagSet) {
 			"will be repeated N times. If > 1 zone specified, nodes will be geo-distributed\n"+
 			"regardless of geo (default [%s])",
 			strings.Join(defaultZones, ",")))
-	flags.BoolVar(&o.preemptible, ProviderName+"-preemptible", false, "use preemptible GCE instances")
+	flags.BoolVar(&o.preemptible, ProviderName+"-preemptible", false,
+		"use preemptible GCE instances (lifetime cannot exceed 24h)")
+	flags.BoolVar(&o.useSpot, ProviderName+"-useSpot", false,
```
Comment (erikgrinaker) [inline pkg/roachprod/vm/gce/gcloud.go:829]: nit: `--gce-use-spot`

Comment (srosenberg) [reply pkg/roachprod/vm/gce/gcloud.go]: updated, it was `--gce-terminateOnMigration` which threw me off.

File: pkg/roachprod/vm/vm.go [hunk 2]
```diff
@@ -73,9 +73,10 @@ type VM struct {
 	CreatedAt time.Time `json:"created_at"`
 	// If non-empty, indicates that some or all of the data in the VM instance
 	// is not present or otherwise invalid.
-	Errors   []error           `json:"errors"`
-	Lifetime time.Duration     `json:"lifetime"`
-	Labels   map[string]string `json:"labels"`
+	Errors      []error           `json:"errors"`
+	Lifetime    time.Duration     `json:"lifetime"`
+	Preemptible bool              `json:"preemptible"`
```
Comment (renatolabs) [inline pkg/roachprod/vm/vm.go:78]: So we make no distinction between `preemptible` and `spot`? Will the distinction ever be useful?

File: pkg/roachprod/vm/gce/gcloud.go [hunk 3]
```diff
@@ -956,6 +974,8 @@ func (p *Provider) Create(
 		// Preemptible instances require the following arguments set explicitly
 		args = append(args, "--maintenance-policy", "TERMINATE")
 		args = append(args, "--no-restart-on-failure")
+	} else if providerOpts.useSpot {
+		args = append(args, "--provisioning-model", "SPOT")
```
Comment (renatolabs) [inline pkg/roachprod/vm/gce/gcloud.go:978]: So the user is supposed to use either `--preemptible` or `use-spot`, right? Should we validate that `!providerOpts.useSpot` when `providerOpts.preemptible`?

**Review by renatolabs — APPROVED** [review]
LGTM, just some questions.

**Review by herkolategan — APPROVED** [review]

**Comment by srosenberg** [comment]
TFTR!

bors r=renatolabs

**Pull request merged**